ui/src/app/kvm/views/KVMDetails/KVMConfiguration/AuthenticationCard/AuthenticationCard.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+
+import {
+  Button,
+  Card,
+  Col,
+  Icon,
+  Link,
+  Row,
+  Textarea,
+} from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
+import FormCard from "app/base/components/FormCard";
+import LabelledList from "app/base/components/LabelledList";
+import { useSendAnalytics } from "app/base/hooks";
+import podSelectors from "app/store/pod/selectors";
+import type { Pod } from "app/store/pod/types";
+import { isPodDetails } from "app/store/pod/utils";
+import type { RootState } from "app/store/root/types";
+
+type Props = {
+  id: Pod["id"];
+};
+
+const AuthenticationCard = ({ id }: Props): JSX.Element | null => {
+  const [showKey, setShowKey] = useState(false);
+  const pod = useSelector((state: RootState) =>
+    podSelectors.getById(state, id)
+  );
+  const sendAnalytics = useSendAnalytics();
+
+  if (isPodDetails(pod)) {
+    const { certificate, power_parameters } = pod;
+
+    // TODO: Handle pods that have not yet generated a certificate.
+    // https://github.com/canonical-web-and-design/app-squad/issues/256
+    if (certificate && power_parameters.certificate && power_parameters.key) {
```

Comment:
If this is not truthy then you don't have an explicit return value.  Might I suggest you adding another conditional like I suggested above?